src/components/dialogs/line-type-catalog/line-type-segment-creation.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import React, { useCallback, useEffect } from 'react';
+import { useWatch } from 'react-hook-form';
+import { gridItem, KilometerAdornment } from '../dialogUtils';
+import EditIcon from '@mui/icons-material/Edit';
+import FloatInput from '../../utils/rhf-inputs/float-input';
+import makeStyles from '@mui/styles/makeStyles';
+import IconButton from '@mui/material/IconButton';
+import Grid from '@mui/material/Grid';
+import {
+    SEGMENT_DISTANCE_VALUE,
+    SEGMENT_TYPE_VALUE,
+    SEGMENT_RESISTANCE,
+    SEGMENT_REACTANCE,
+    SEGMENT_SUSCEPTANCE,
+} from '../../utils/field-constants';
+import { ReadOnlyInput } from '../../utils/rhf-inputs/read-only/read-only-input';
+import PropTypes from 'prop-types';
+
+const useStyles = makeStyles((theme) => ({
+    simpleLabel: {
+        paddingTop: theme.spacing(1),
+    },
+}));
+
+const LineTypeSegmentCreation = ({
+    name,
+    index,
+    onEditButtonClick,
+    onSegmentDistanceChange,
+}) => {
+    const classes = useStyles();
+
+    const segmentDistanceField = (
```

Comment:
I'm confused with this PR, sometimes you create const to store inputs, sometimes you create them directly in the render block

I think we should stick to one way to do it

Reply:
same thing in "line-type-segment-form.js", it would greatly reduce the JSX complexity

Reply:
Done.